cmd/dagger/module_inspect.go

```diff
@@ -426,11 +417,6 @@ func (m *moduleDef) GetObjectFunction(objectName, functionName string) (*modFunc
 }
 
 func (m *moduleDef) GetFunction(fp functionProvider, functionName string) (*modFunction, error) {
-	// This avoids an issue with module constructors overriding core functions.
-	// See https://github.com/dagger/dagger/issues/9122
-	if m.HasModule() && fp.ProviderName() == "Query" && m.MainObject.AsObject.Constructor.CmdName() == functionName {
-		return m.MainObject.AsObject.Constructor, nil
-	}
```

Comment:
@helderco was this not working as described in the comment?

Reply:
At the time the "Query" object didn't include the module's constructor, it was only core, before loading the module. I guess that was fixed when Erik refactored module loading. Now, I just call `loadTypeDefs` on every function under `Query` when the module is inspected, which includes the module's constructor so this is no longer needed.